common/logger/src/struct_log.rs

```diff
@@ -284,3 +295,66 @@ impl FileStructLogThread {
         }
     }
 }
+
+/// Sink that streams all structured logs to an address through UDP protocol
+struct UDPStructLog {
+    sender: SyncSender<StructuredLogEntry>,
+}
+
+impl UDPStructLog {
+    /// Creates new UDPStructLog and starts async thread to send results
+    pub fn start_new(udp_address: String) -> io::Result<Self> {
+        let (sender, receiver) = mpsc::sync_channel(1_024);
+        let socket = UdpSocket::bind(udp_address.clone()).expect("couldn't bind to address");
+        let sink_thread = UDPStructLogThread {
+            receiver,
+            socket,
+            udp_address,
+        };
+        thread::spawn(move || sink_thread.run());
+        Ok(Self { sender })
+    }
+}
+
+impl StructLogSink for UDPStructLog {
+    fn send(&self, entry: StructuredLogEntry) {
+        if let Err(e) = self.sender.try_send(entry) {
+            // Use log crate macro to avoid generation of structured log in this case
+            // Otherwise we will have infinite loop
+            log::error!("Failed to send structured log: {}", e);
+        }
+    }
+}
+
+struct UDPStructLogThread {
+    receiver: Receiver<StructuredLogEntry>,
+    socket: UdpSocket,
+    udp_address: String,
+}
+
+impl UDPStructLogThread {
+    pub fn run(self) {
+        for entry in self.receiver {
+            let json = match serde_json::to_value(entry) {
+                Err(e) => {
+                    log::error!("Failed to serialize struct log entry: {}", e);
+                    continue;
+                }
+                Ok(json) => json,
+            };
+            match self
+                .socket
+                .send_to(json.to_string().as_bytes(), self.udp_address.clone())
+            {
+                Ok(_) => {
+                    println!("Sent message {}", json.to_string());
```

Comment:
We probably don't want to actually have println! here on every message - this will go to stdout and consume space